stl/inc/algorithm

```diff
@@ -2422,6 +2422,133 @@ namespace ranges {
     };
 
     inline constexpr _Search_n_fn search_n{_Not_quite_object::_Construct_tag{}};
+
+#if _HAS_CXX23
+    // VARIABLE ranges::starts_with
+    class _Starts_with_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2,
+            class _Pr = ranges::equal_to, class _Pj1 = identity, class _Pj2 = identity>
+            requires indirectly_comparable<_It1, _It2, _Pr, _Pj1, _Pj2>
+        _NODISCARD constexpr bool operator()(_It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2, _Pr _Pred = {},
+            _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            // clang-format on
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+
+            if constexpr (sized_sentinel_for<_Se1, _It1> && sized_sentinel_for<_Se2, _It2>) {
+                const iter_difference_t<_It1> _Count1 = _Last1 - _First1;
+                const iter_difference_t<_It2> _Count2 = _Last2 - _First2;
+                if (_Count2 > _Count1) {
+                    return false;
+                }
+
+                const auto _Result = _Mismatch_n(_STD move(_First1), _STD move(_First2),

```

Comment:
We should use `equal` with sized sentinels and contiguous iterators to have `memcmp` optimization.

Reply:
That is a good idea 👍